### ISSUE TITLE
setState use call inside of <Animated.Code> memory leak

### ISSUE THREAD

**Issue by tangtai** [software-mansion/react-native-reanimated#332]
i try to update my state from animated value, and i subscribe to the animated value and update my state, but it causes memory leak, it updates every 10ms even if the animated value is not changed, i only want the state to be updated when there is a change in the animated value, where i can call from a shouldcomponentUpdate(prevState), but it seems like it can't retrive the  animated value outside of the render. i wonder what is the best way to do this. here is my code:
`console.log`
```
const {call} = Animated

class SomeComponent extends Component {
constructor(props) {
    super(props);
    this._transY = new Value(0);
    this.state = { height: "", };
  }

 updateHeight = value => {
    requestAnimationFrame(() => {
      this.setState({ newHeight: value });
    });
  };

render(){
   return (
   <Animated.View>
       <Animated.Code>{() => call([this._transY], this.updateHeight)}</Animated.Code>
   </Animated.View>
 )
}

```

**Comment by wcandillon** [comment]
I can confirm a similar memory leak with `<Animated.Code>`. It doesn't happen if I use `useCode`.

**Comment by wcandillon** [comment]
I would rename this issue to "<Animated.Code> memory leak with call()"

**Comment by wcandillon** [comment]
I've been experiencing leaking behavior as well with `<Animated.Code>`, doesn't happen with useCode. in the case off `call()` could it be possible that the listener/callback doesn't get cleaned up?

**Comment by tangtai** [comment]
so the Animated value is getting updated from an Interactive.View, event if the Value is not being updated, the listener continous updating the value. the current work around i did is to only allow the setState being called when the Interactive.view is stopped with onStop method. and adding debouce to update the value in redux, so the redux is not updated 20 times after each onStop method being called. it feels hackish but it works.

**Comment by WoLewicki** [comment]
Is it a complementary example or should there be sth added in order for the bug to be spotted? If so, please add the necessary code.

**Comment by wcandillon** [comment]
@osdnk would it make sense to rewrite `Animated.Code` as a functional component with `useCode` in order to ensure that the behavior is consistent between the two? If yes, I'd be happy to make a PR.

**Comment by jakub-gonet** [comment]
Implemented @wcandillon suggestion in #952.
